bigframes/core/nodes.py

```diff
@@ -398,6 +415,11 @@ def relation_ops_created(self) -> int:
         # Assume worst case, where readgbq actually has baked in analytic operation to generate index
         return 3
 
+    @property
+    def supports_fast_head(self) -> bool:
+        # TODO: Be more lenient for small tables, or those clustered on non-sequential order key
```

Comment:
+1, `LIMIT` clause is quite cheap on clustered tables.

Reply:
From what I can tell, this is not actually the case yet. All the files still get dispatched for ORDER BY+LIMIT, and so full cost is paid. Only filter over materialized row numbers is granting significant saving from my experiements. This may change in time.